### PR TITLE
fix(ui): advance search ui updates

### PR DESCRIPTION
**Pull request by chirag-madlani**
fix quick filters not working

### Describe your changes :

I worked on the updating ui for advance search modal
- Fixed broken filters on explore page

#
### Type of change :

- [x] Bug fix

#
### Frontend Preview (Screenshots) :
Screenshot 2022-11-22 at 7 21 38 PM

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Explore/AdvanceSearchModal.component.tsx [hunk 1]
```diff
@@ -28,18 +29,39 @@ export const AdvancedSearchModal: FunctionComponent<Props> = ({
...
+      closable={false}
+      footer={
+        <Space className="justify-between w-full">
+          <Button
+            className="float-right"
+            size="small"
+            onClick={handleAdvanceSearchReset}>
+            Reset
+          </Button>
+          <div>
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Explore/AdvanceSearchModal.component.tsx:48]: I would recommend using fragments.

File: openmetadata-ui/src/main/resources/ui/src/components/Explore/Explore.component.tsx [hunk 2]
```diff
@@ -285,6 +286,14 @@ const Explore: React.FC<ExploreProps> = ({
...
       }
     });
+
+    onChangeAdvancedSearchQueryFilter(
+      isEmpty(term)
+        ? undefined
+        : {
+            query: { bool: { must: [{ term }] } },
+          }
+    );
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Explore/Explore.component.tsx:296]: Do we need this change every time on the `selectedQuickFilters` change?

File: openmetadata-ui/src/main/resources/ui/src/styles/components/react-awesome-query.less [hunk 3]
```diff
@@ -24,6 +24,13 @@
 
   & > .group-or-rule-container.group-container {
     & > .group.group-or-rule {
+      .group--children {
+        .group-or-rule-container.group-container:first-child {
+          .group--header .action.action--DELETE {
+            display: none;
+          }
+        }
+      }
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/styles/components/react-awesome-query.less:33]: Do we want this only for the first delete option?

File: openmetadata-ui/src/main/resources/ui/src/styles/components/react-awesome-query.less [hunk 4]
```diff
@@ -208,3 +215,38 @@
...
+}
+.rule-container + .rule-container .ant-btn-group {
+  visibility: visible;
+}
+
+.rule--field {
+  width: 180px;
+
+  .ant-select {
+    width: 100% !important;
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/styles/components/react-awesome-query.less:230]: See if we can avoid the `!important` here.

File: openmetadata-ui/src/main/resources/ui/src/components/Explore/AdvanceSearchModal.component.tsx (hunk 1 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/Explore/AdvanceSearchModal.component.tsx:48]: fragment won't add anything to DOM and I want to add justifyBetween for these two elements. Let me know if Fragment will work here

File: openmetadata-ui/src/main/resources/ui/src/styles/components/react-awesome-query.less (hunk 3 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/styles/components/react-awesome-query.less:33]: Yes, We want to restrict user from deleting one and only group which will leads to uncertain UI

File: openmetadata-ui/src/main/resources/ui/src/styles/components/react-awesome-query.less (hunk 4 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/styles/components/react-awesome-query.less:230]: I am still figuring out a way to set width via config, but till then `!important` is the option I am left with

File: openmetadata-ui/src/main/resources/ui/src/components/Explore/AdvanceSearchModal.component.tsx (hunk 1 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/Explore/AdvanceSearchModal.component.tsx:48]: Make sense, fragment will not good fit in that case.

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**